eppo_core/src/precomputed_assignments.rs

```diff
@@ -0,0 +1,86 @@
+use crate::ufc::{Assignment, AssignmentFormat, Environment, VariationType};
+use crate::{Attributes, Configuration, Str};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+// Request
+#[derive(Debug, Deserialize)]
+pub struct PrecomputedAssignmentsServiceRequestBody {
+    pub subject_key: String,
+    pub subject_attributes: Arc<Attributes>,
+    // TODO: Add bandit actions
+    // #[serde(rename = "banditActions")]
+    // #[serde(skip_serializing_if = "Option::is_none")]
+    // bandit_actions: Option<HashMap<String, serde_json::Value>>,
+}
+
+// Response
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FlagAssignment {
+    pub allocation_key: Str,
+    pub variation_key: Str,
+    pub variation_type: VariationType,
+    pub variation_value: serde_json::Value,
+    /// Additional user-defined logging fields for capturing extra information related to the
+    /// assignment.
+    #[serde(flatten)]
+    pub extra_logging: HashMap<String, String>,
+    pub do_log: bool,
+}
+
+impl FlagAssignment {
+    pub fn try_from_assignment(assignment: Assignment) -> Option<Self> {
+        // WARNING! There is a problem here. The event is only populated for splits
```

Comment:
@rasendubi @sameerank I believe this is a problem as described here: https://github.com/Eppo-exp/eppo-multiplatform/blob/main/eppo_core/src/ufc/compiled_flag_config.rs#L231

During compilation, an optimization step to reduce the configuration, `event` is only populated for splits with `do_log == True`. 

A possible solution is to produce a configuration that is "not optimized" for use in this function so that we can access `do_log`

Reply:
I still think its a problem because we want the precomputed server to return evaluations for all flags and `do_log` is true or false depending on if the user wants them log, but they still need to be present. The flag will be present in the response which changes the behavior of the SDK / application instead of now the flag will not be present and the default value will be used. Am I understanding that correctly?